webapp/channels/src/components/post_view/post_flag_icon/post_flag_icon.tsx

```diff
@@ -19,115 +19,97 @@ export type Actions = {
     unflagPost: typeof unflagPost;
 }
 
-interface Props {
+type Props = {
     location?: keyof typeof Locations;
     postId: string;
     isFlagged: boolean;
     actions: Actions;
 }
 
-interface State {
-    a11yActive: boolean;
-}
-
-export default class PostFlagIcon extends React.PureComponent<Props, State> {
-    static defaultProps = {
-        location: Locations.CENTER,
-    };
-
-    private buttonRef: React.RefObject<HTMLButtonElement>;
-
-    constructor(props: Props) {
-        super(props);
-
-        this.buttonRef = React.createRef();
-
-        this.state = {
-            a11yActive: false,
-        };
-    }
-
-    componentDidMount() {
-        if (this.buttonRef.current) {
-            this.buttonRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
-            this.buttonRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
-        }
-    }
-    componentWillUnmount() {
-        if (this.buttonRef.current) {
-            this.buttonRef.current.removeEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
-            this.buttonRef.current.removeEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
-        }
-    }
-
-    componentDidUpdate() {
-        if (this.state.a11yActive && this.buttonRef.current) {
-            this.buttonRef.current.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
-        }
-    }
-
-    handlePress = (e: React.MouseEvent) => {
+const PostFlagIcon = ({
+    actions: {
+        flagPost,
+        unflagPost,
+    },
+    isFlagged,
+    postId,
+    location = Locations.CENTER,
+}: Props) => {
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    const [a11yActive, setA11yActive] = useState(false);
+
+    const handlePress = useCallback((e: React.MouseEvent) => {
         e.preventDefault();
 
-        const {
-            actions,
-            isFlagged,
-            postId,
-        } = this.props;
-
         if (isFlagged) {
-            actions.unflagPost(postId);
+            unflagPost(postId);
         } else {
-            actions.flagPost(postId);
+            flagPost(postId);
         }
-    };
+    }, [flagPost, unflagPost, postId, isFlagged]);
 
-    handleA11yActivateEvent = () => {
-        this.setState({a11yActive: true});
-    };
-
-    handleA11yDeactivateEvent = () => {
-        this.setState({a11yActive: false});
-    };
+    useEffect(() => {
+        const handleA11yActivateEvent = () => {
+            setA11yActive(true);
+        };
+        const handleA11yDeactivateEvent = () => {
+            setA11yActive(false);
+        };
```

Comment:
Any reasons why these are anonymous functions stored in a variable instead of plain functions?

Reply:
If you mean why use `const foo = () => {...}` instead of using just `() => {...}` where needed, it is because I want to send the same reference to both add and remove listener.

If you mean why use `const foo = () => {...}` instead of `func foo () {...}`, no reason at all. Not sure if there is any real difference between both options.

Reply:
yes i meant latter, there is no difference anyways. We don't want to set expectations for contributors to always use anonymous functions instead of plain functions.